### DISCUSSION TITLE
Peppol production - Detection error (FRTEST): Validation of subject principal(CN) failed.

### DISCUSSION THREAD

**Discussion by tinhien11** [OxalisCommunity/oxalis#700] [General]
Hi, I'm testing in production. 
Oxalis runs well for message reception and submission in development, but I'm facing issues in production.

version.oxalis.as4: 6.7.0
version.java: 11.0.16
mode: PRODUCTION

This is the error log:

13:32:35.606 [main] INFO  n.o.v.peppol.security.ModeDetector - Detected mode: PRODUCTION. 
13:32:30.100 [main] INFO  n.o.v.peppol.security.ModeDetector - Detection error (LOCAL): Certificate should be self-signed.  
13:32:30.008 [main] INFO  n.o.v.peppol.security.ModeDetector - Detection error (TEST): Validation of subject principal(CN) failed.  
13:32:29.906 [main] INFO  n.o.v.peppol.security.ModeDetector - Detection error (FRTEST): Validation of subject principal(CN) failed.

13:31:59,100 |-INFO in ch.qos.logback.classic.util.ContextInitializer@6973bf95 - ch.qos.logback.classic.util.DefaultJoranConfigurator.configure() call lasted 4395 milliseconds. ExecutionStatus=DO_NOT_INVOKE_NEXT_IF_ANY
13:31:59,014 |-INFO in ch.qos.logback.core.model.processor.DefaultProcessor@61dd025 - End of configuration.
13:31:59,013 |-INFO in ch.qos.logback.core.model.processor.AppenderRefModelHandler - Attaching appender named [STDOUT] to Logger[ROOT]
13:31:58,999 |-INFO in ch.qos.logback.classic.model.processor.RootLoggerModelHandler - Setting level of ROOT logger to INFO
13:31:58,511 |-INFO in ch.qos.logback.core.model.processor.ImplicitModelHandler - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
13:31:58,199 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
13:31:57,801 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - Processing appender named [STDOUT]

Any pointers on what I should check? Thanks so much

**Comment by aaron-kumar** [comment]
@tinhien11 If you are running Oxalis in production with Production certificate then as per log correct "production" certificate mode is detected _"Detected mode: PRODUCTION"_ . This is correct behavior 
I don't think you are using France Peppol certificate - "Detection error (FRTEST): Validation of subject principal(CN) failed." 

Additionally, I can't see any error in logs. Please share details log so that somebody can help you. 

Note: Since this is Not an issue so converting it into discussion for further collaboration/discussion.

**Reply by tinhien11** [reply]
Apologies for the delayed response. 
I believe that the message "Detection error (FRTEST): Validation of subject principal (CN) failed" is not an issue. 
I will carefully check it again and reply here.

**Reply by tinhien11** [reply]
Thank you for your help. It works fine now. The issue is being sent to the wrong Peppol ID.
